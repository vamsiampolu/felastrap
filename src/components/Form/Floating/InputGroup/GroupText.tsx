import { useFela } from "react-fela";
import { GroupText, type GroupTextProps } from "../../InputGroup/GroupText";

const floatingRule = () => ({
  height: "calc(3.5rem + 2px)",
  minHeight: "calc(3.5rem + 2px)",
});

export const FloatingGroupText = ({
  className = "",
  children,
  ...props
}: GroupTextProps) => {
  const { css } = useFela();
  return (
    <GroupText
      className={`${css(
        floatingRule,
      )} floating-input-group-text ${className}`.trim()}
      {...props}
    >
      {children}
    </GroupText>
  );
};
