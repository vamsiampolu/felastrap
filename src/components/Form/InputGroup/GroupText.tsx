import { ComponentPropsWithoutRef, useContext } from "react";
import { useFela } from "react-fela";
import { InputGroupContext } from "./InputGroupContext";
import { Size, StyleProps } from "../../common";

type RuleProps = StyleProps<{ size: Size }>;

const inputGroupTextRule = () => ({
  display: "flex",
  alignItems: "center",
  padding: "0.375rem 0.75rem",
  fontSize: "1rem",
  fontWeight: 400,
  lineHeight: 1.5,
  color: "#212529",
  textAlign: "center",
  whiteSpace: "nowrap",
  backgroundColor: "#f8f9fa",
  border: "1px solid #dee2e6",
  borderRadius: "0.375rem",
});

const inputGroupSmallTextRule = ({ size }: RuleProps) =>
  size === "small"
    ? {
        padding: "0.25rem 0.5rem",
        fontSize: "0.875rem",
        borderRadius: "0.25rem",
      }
    : {};

const inputGroupLargeTextRule = ({ size }: RuleProps) =>
  size === "large"
    ? {
        padding: "0.5rem 1rem",
        fontSize: "1.25rem",
        borderRadius: "0.5rem",
      }
    : {};

export type Children = ComponentPropsWithoutRef<"span">["children"];
export type SpanProps = Omit<
  ComponentPropsWithoutRef<"span">,
  "children" | "className"
>;

export type GroupTextProps = {
  children: Children;
  className?: string;
} & SpanProps;

export const GroupText = ({
  className = "",
  children,
  ...props
}: GroupTextProps) => {
  const { size = "medium" } = useContext(InputGroupContext);
  const { css } = useFela({ size });

  return (
    <span
      data-testid="input-group-text"
      className={`${css([
        inputGroupTextRule,
        inputGroupSmallTextRule,
        inputGroupLargeTextRule,
      ])} input-group-text ${className}`}
      {...props}
    >
      {children}
    </span>
  );
};
