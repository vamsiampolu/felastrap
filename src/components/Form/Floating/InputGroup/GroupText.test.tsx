import { IRenderer } from "fela";
import { FloatingGroupText } from "./GroupText";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../../test-helpers";
import { render, screen } from "@testing-library/react";

describe("components/Form/Floating/InputGroup/GroupText", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should display a group text", async () => {
    const component = (
      <FloatingGroupText>Floating Group Text</FloatingGroupText>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("input-group-text")).toBeInTheDocument();

    expect(screen.queryByTestId("input-group-text")).toHaveClass(
      "floating-input-group-text",
    );
    expect(screen.queryByTestId("input-group-text")).toHaveTextContent(
      "Floating Group Text",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
