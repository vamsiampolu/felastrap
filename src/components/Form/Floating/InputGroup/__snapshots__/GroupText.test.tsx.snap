// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form/Floating/InputGroup/GroupText should display a group text 1`] = `
".a {
  height: calc(3.5rem + 2px);
}
.b {
  min-height: calc(3.5rem + 2px);
}
.c {
  display: flex;
}
.d {
  align-items: center;
}
.e {
  font-size: 1rem;
}
.f {
  font-weight: 400;
}
.g {
  line-height: 1.5;
}
.h {
  color: #212529;
}
.i {
  text-align: center;
}
.j {
  white-space: nowrap;
}
.k {
  background-color: #f8f9fa;
}
.l {
  border-top-left-radius: 0.375rem;
}
.m {
  border-top-right-radius: 0.375rem;
}
.n {
  border-bottom-right-radius: 0.375rem;
}
.o {
  border-bottom-left-radius: 0.375rem;
}
.p {
  border-top-width: 1px;
}
.q {
  border-right-width: 1px;
}
.r {
  border-bottom-width: 1px;
}
.s {
  border-left-width: 1px;
}
.t {
  border-top-style: solid;
}
.u {
  border-right-style: solid;
}
.v {
  border-bottom-style: solid;
}
.w {
  border-left-style: solid;
}
.x {
  border-top-color: #dee2e6;
}
.y {
  border-right-color: #dee2e6;
}
.z {
  border-bottom-color: #dee2e6;
}
.ab {
  border-left-color: #dee2e6;
}
.ac {
  padding-top: 0.375rem;
}
.ae {
  padding-right: 0.75rem;
}
.af {
  padding-bottom: 0.375rem;
}
.ag {
  padding-left: 0.75rem;
}


<span
  data-testid="input-group-text"
  class="ab ac ae af ag c d e f g h i j k l m n o p q r s t u v w x y z input-group-text a b floating-input-group-text"
  >Floating Group Text</span
>;
"
`;
